// ra-supabase-core/lib/useResetPassword.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.useResetPassword = void 0;
var ra_core_1 = require("ra-core");
var react_query_1 = require("@tanstack/react-query");
/**
 * This hook returns a function to call in order to reset a user password on Supabase.
 *
 * @example
 * import { useSupabaseAccessToken } from 'ra-supabase-core';
 *
 * const ResetPasswordPage = () => {
 *     const [resetPassword] = useResetPassword();
 *
 *     const handleSubmit = event => {
 *         resetPassword({
 *             email: event.currentTarget.elements.email.value,
 *         });
 *     };
 *
 *     return (
 *         <form onSubmit={handleSubmit}>
 *             <label for="email">Email:</label>
 *             <input id="email" name="email" type="email" />
 *             <button type="submit">Reset password</button>
 *         </form>
 *     );
 * };
 **/
var useResetPassword = function (options) {
    var notify = (0, ra_core_1.useNotify)();
    var redirect = (0, ra_core_1.useRedirect)();
    var authProvider = (0, ra_core_1.useAuthProvider)();
    if (authProvider == null) {
        throw new Error('No authProvider found. Did you forget to set up an AuthProvider on the <Admin> component?');
    }
    if (authProvider.resetPassword == null) {
        throw new Error('The setPassword() method is missing from the AuthProvider although it is required. You may consider adding it');
    }
    var _a = options || {}, _b = _a.onSuccess, onSuccess = _b === void 0 ? function () {
        redirect('/login');
        notify('ra-supabase.auth.password_reset', { type: 'info' });
    } : _b, _c = _a.onError, onError = _c === void 0 ? function (error) { return notify(error.message, { type: 'error' }); } : _c;
    var mutation = (0, react_query_1.useMutation)({
        mutationFn: function (params) {
            return authProvider.resetPassword(params);
        },
        onSuccess: onSuccess,
        onError: onError,
        retry: false,
    });
    return [mutation.mutate, mutation];
};
exports.useResetPassword = useResetPassword;
