// ra-supabase-ui-materialui/lib/ForgotPasswordForm.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ForgotPasswordForm = void 0;
var material_1 = require("@mui/material");
var ra_core_1 = require("ra-core");
var ra_supabase_core_1 = require("ra-supabase-core");
var ra_ui_materialui_1 = require("ra-ui-materialui");
var React = __importStar(require("react"));
/**
 * A component that renders a form for resetting the user password.
 */
var ForgotPasswordForm = function () {
    var notify = (0, ra_core_1.useNotify)();
    var translate = (0, ra_core_1.useTranslate)();
    var resetPassword = (0, ra_supabase_core_1.useResetPassword)({
        onError: function (error) {
            notify(typeof error === 'string'
                ? error
                : typeof error === 'undefined' || !error.message
                    ? 'ra.auth.sign_in_error'
                    : error.message, {
                type: 'warning',
                messageArgs: {
                    _: typeof error === 'string'
                        ? error
                        : error && error.message
                            ? error.message
                            : undefined,
                },
            });
        },
    })[0];
    var submit = function (values) {
        return resetPassword({
            email: values.email,
        });
    };
    return (React.createElement(Root, { onSubmit: submit },
        React.createElement("div", { className: SupabaseLoginFormClasses.container },
            React.createElement(material_1.Stack, { spacing: 1 },
                React.createElement(material_1.Typography, { variant: "h5", textAlign: "center" }, translate('ra-supabase.reset_password.forgot_password', { _: 'Forgot password?' })),
                React.createElement(material_1.Typography, { variant: "body2", color: "GrayText", textAlign: "center" }, translate('ra-supabase.reset_password.forgot_password_details', {
                    _: 'Enter your email to receive a reset password link.',
                }))),
            React.createElement("div", { className: SupabaseLoginFormClasses.input },
                React.createElement(ra_ui_materialui_1.TextInput, { source: "email", label: translate('ra.auth.email', {
                        _: 'Email',
                    }), autoComplete: "email", fullWidth: true, validate: (0, ra_core_1.required)() }))),
        React.createElement(material_1.CardActions, { sx: { flexDirection: 'column', gap: 1 } },
            React.createElement(ra_ui_materialui_1.SaveButton, { variant: "contained", type: "submit", className: SupabaseLoginFormClasses.button, label: translate('ra.action.reset_password', {
                    _: 'Reset password',
                }), icon: React.createElement(React.Fragment, null) }),
            React.createElement(ra_ui_materialui_1.Link, { to: "/login", variant: "body2" }, translate('ra-supabase.auth.back_to_login', {
                _: 'Back to login page',
            })))));
};
exports.ForgotPasswordForm = ForgotPasswordForm;
var PREFIX = 'RaSupabaseForgotPasswordForm';
var SupabaseLoginFormClasses = {
    container: "".concat(PREFIX, "-container"),
    input: "".concat(PREFIX, "-input"),
    button: "".concat(PREFIX, "-button"),
};
var Root = (0, material_1.styled)(ra_core_1.Form, {
    name: PREFIX,
    overridesResolver: function (props, styles) { return styles.root; },
})(function (_a) {
    var _b;
    var theme = _a.theme;
    return (_b = {},
        _b["& .".concat(SupabaseLoginFormClasses.container)] = {
            padding: '0 1em 0 1em',
        },
        _b["& .".concat(SupabaseLoginFormClasses.input)] = {
            marginTop: '1em',
        },
        _b["& .".concat(SupabaseLoginFormClasses.button)] = {
            width: '100%',
        },
        _b);
});
