// ra-supabase-ui-materialui/lib/LoginForm.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LoginForm = void 0;
var material_1 = require("@mui/material");
var ra_core_1 = require("ra-core");
var ra_ui_materialui_1 = require("ra-ui-materialui");
var React = __importStar(require("react"));
var ForgotPasswordPage_1 = require("./ForgotPasswordPage");
/**
 * A component that renders a form to login to the application with an email and password.
 */
var LoginForm = function (_a) {
    var disableForgotPassword = _a.disableForgotPassword, props = __rest(_a, ["disableForgotPassword"]);
    var login = (0, ra_core_1.useLogin)();
    var notify = (0, ra_core_1.useNotify)();
    var translate = (0, ra_core_1.useTranslate)();
    var submit = function (values) {
        return login(values).catch(function (error) {
            notify(typeof error === 'string'
                ? error
                : typeof error === 'undefined' || !error.message
                    ? 'ra.auth.sign_in_error'
                    : error.message, {
                type: 'warning',
                messageArgs: {
                    _: typeof error === 'string'
                        ? error
                        : error && error.message
                            ? error.message
                            : undefined,
                },
            });
        });
    };
    return (React.createElement(Root, __assign({ onSubmit: submit }, props),
        React.createElement("div", { className: SupabaseLoginFormClasses.container },
            React.createElement("div", { className: SupabaseLoginFormClasses.input },
                React.createElement(ra_ui_materialui_1.TextInput, { autoFocus: true, source: "email", type: "email", label: translate('ra-supabase.auth.email', {
                        _: 'Email',
                    }), fullWidth: true, validate: (0, ra_core_1.required)() })),
            React.createElement("div", null,
                React.createElement(ra_ui_materialui_1.PasswordInput, { source: "password", label: translate('ra.auth.password', {
                        _: 'Password',
                    }), autoComplete: "current-password", fullWidth: true, validate: (0, ra_core_1.required)() }))),
        React.createElement(material_1.CardActions, { sx: { flexDirection: 'column', gap: 1 } },
            React.createElement(ra_ui_materialui_1.SaveButton, { variant: "contained", type: "submit", className: SupabaseLoginFormClasses.button, label: translate('ra.auth.sign_in'), icon: React.createElement(React.Fragment, null) }),
            !disableForgotPassword ? (React.createElement(ra_ui_materialui_1.Link, { to: ForgotPasswordPage_1.ForgotPasswordPage.path, variant: "body2" }, translate('ra-supabase.auth.forgot_password', {
                _: 'Forgot password?',
            }))) : null)));
};
exports.LoginForm = LoginForm;
var PREFIX = 'RaSupabaseLoginForm';
var SupabaseLoginFormClasses = {
    container: "".concat(PREFIX, "-container"),
    input: "".concat(PREFIX, "-input"),
    button: "".concat(PREFIX, "-button"),
};
var Root = (0, material_1.styled)(ra_core_1.Form, {
    name: PREFIX,
    overridesResolver: function (props, styles) { return styles.root; },
})(function (_a) {
    var _b;
    var theme = _a.theme;
    return (_b = {},
        _b["& .".concat(SupabaseLoginFormClasses.container)] = {
            padding: '0 1em 1em 1em',
        },
        _b["& .".concat(SupabaseLoginFormClasses.input)] = {
            marginTop: '1em',
        },
        _b["& .".concat(SupabaseLoginFormClasses.button)] = {
            width: '100%',
        },
        _b);
});
