// ra-supabase-ui-materialui/lib/SocialAuthButton.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkosButton = exports.TwitterButton = exports.TwitchButton = exports.SpotifyButton = exports.SlackButton = exports.NotionButton = exports.LinkedInButton = exports.KeycloakButton = exports.GoogleButton = exports.GithubButton = exports.GitlabButton = exports.FacebookButton = exports.DiscordButton = exports.BitbucketButton = exports.AzureButton = exports.AppleButton = exports.SocialAuthButton = void 0;
var React = __importStar(require("react"));
var material_1 = require("@mui/material");
var ra_core_1 = require("ra-core");
var icons_1 = require("./icons");
var SocialAuthButton = function (_a) {
    var provider = _a.provider, redirectTo = _a.redirect, props = __rest(_a, ["provider", "redirect"]);
    var login = (0, ra_core_1.useLogin)();
    var notify = (0, ra_core_1.useNotify)();
    var handleClick = function () {
        login({ provider: provider }, redirectTo !== null && redirectTo !== void 0 ? redirectTo : window.location.toString()).catch(function (error) {
            // The authProvide always reject for OAuth login but there will be no error
            // if the call actually succeeds. This is to avoid react-admin redirecting
            // immediately to the provided redirect prop before users are redirected to
            // the OAuth provider.
            if (error) {
                notify(error.message, { type: 'error' });
            }
        });
    };
    return (React.createElement(material_1.Button, __assign({ onClick: handleClick, variant: "contained", size: "medium", color: "inherit" }, props)));
};
exports.SocialAuthButton = SocialAuthButton;
var AppleButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Apple',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.AppleIcon, null), provider: "apple" }, props), label));
};
exports.AppleButton = AppleButton;
var AzureButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Azure',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.AzureIcon, null), provider: "azure" }, props), label));
};
exports.AzureButton = AzureButton;
var BitbucketButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Bitbucket',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.BitbucketIcon, null), provider: "bitbucket" }, props), label));
};
exports.BitbucketButton = BitbucketButton;
var DiscordButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Discord',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.DiscordIcon, null), provider: "discord" }, props), label));
};
exports.DiscordButton = DiscordButton;
var FacebookButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Facebook',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.FacebookIcon, null), provider: "facebook" }, props), label));
};
exports.FacebookButton = FacebookButton;
var GitlabButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Gitlab',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.GitlabIcon, null), provider: "gitlab" }, props), label));
};
exports.GitlabButton = GitlabButton;
var GithubButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Github',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.GithubIcon, null), provider: "github" }, props), label));
};
exports.GithubButton = GithubButton;
var GoogleButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Google',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.GoogleIcon, null), provider: "google" }, props), label));
};
exports.GoogleButton = GoogleButton;
var KeycloakButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Keycloak',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.KeycloakIcon, null), provider: "keycloak" }, props), label));
};
exports.KeycloakButton = KeycloakButton;
var LinkedInButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'LinkedIn',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.LinkedinIcon, null), provider: "linkedin" }, props), label));
};
exports.LinkedInButton = LinkedInButton;
var NotionButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Notion',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.NotionIcon, null), provider: "notion" }, props), label));
};
exports.NotionButton = NotionButton;
var SlackButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Slack',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.SlackIcon, null), provider: "slack" }, props), label));
};
exports.SlackButton = SlackButton;
var SpotifyButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Spotify',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.SpotifyIcon, null), provider: "spotify" }, props), label));
};
exports.SpotifyButton = SpotifyButton;
var TwitchButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Twitch',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.TwitchIcon, null), provider: "twitch" }, props), label));
};
exports.TwitchButton = TwitchButton;
var TwitterButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'Twitter',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.TwitterIcon, null), provider: "twitter" }, props), label));
};
exports.TwitterButton = TwitterButton;
var WorkosButton = function (props) {
    var translate = (0, ra_core_1.useTranslate)();
    var label = translate('ra-supabase.auth.sign_in_with', {
        provider: 'WorkOS',
    });
    return (React.createElement(exports.SocialAuthButton, __assign({ startIcon: React.createElement(icons_1.WorkosIcon, null), provider: "workos" }, props), label));
};
exports.WorkosButton = WorkosButton;
